base/compiler/ssair/passes.jl

```diff
@@ -1,3 +1,110 @@
+function compact_exprtype(compact, value)
+    if isa(value, Union{SSAValue, OldSSAValue})
+        return types(compact)[value]
+    elseif isa(value, Argument)
+        return compact.ir.argtypes[value.n]
+    end
+    exprtype(value, compact.ir, compact.ir.mod)
+end
+
+function getfield_elim_pass!(ir::IRCode)
```

Comment:
Could use some more comments to split up the function into the stages of the algorithm. It's a little hard to see the high-level view right now, before I jump into reviewing the line-by-line details.